src/dashboard/infrastructure/scores/score-link-collector.php

```diff
@@ -0,0 +1,49 @@
+<?php
+// phpcs:disable Yoast.NamingConventions.NamespaceName.TooLong
+// phpcs:disable Yoast.NamingConventions.NamespaceName.MaxExceeded
+namespace Yoast\WP\SEO\Dashboard\Infrastructure\Scores;
+
+use Yoast\WP\SEO\Dashboard\Domain\Content_Types\Content_Type;
+use Yoast\WP\SEO\Dashboard\Domain\Scores\Scores_Interface;
+use Yoast\WP\SEO\Dashboard\Domain\Taxonomies\Taxonomy;
+
+/**
+ * Getting links for scores.
+ */
+class Score_Link_Collector {
+
+	/**
+	 * Builds the view link of the score.
+	 *
+	 * @param Scores_Interface $score_name   The name of the score.
+	 * @param Content_Type     $content_type The content type.
+	 * @param Taxonomy|null    $taxonomy     The taxonomy of the term we might be filtering.
+	 * @param int|null         $term_id      The ID of the term we might be filtering.
+	 *
+	 * @return string The view link of the score.
+	 */
+	public function get_view_link( Scores_Interface $score_name, Content_Type $content_type, ?Taxonomy $taxonomy, ?int $term_id ): ?string {
+		$posts_page = \admin_url( 'edit.php' );
+		$args       = [
+			'post_status'                  => 'publish',
+			'post_type'                    => $content_type->get_name(),
+			$score_name->get_filter_key()  => $score_name->get_filter_value(),
+		];
+
+		if ( $taxonomy === null || $term_id === null ) {
```

Comment:
You might put this first before ever doing anything with the admin_url and building the args array

Reply:
I don't get your point here 🤔 I'm already using the `admin_url` and the args array even when this if is true, so I don't see it.

Reply:
Never mind I cannot read.